QuadPlane TVBS: set plane.nav_roll/pitch in QACRO mode

kd0aij opened the pull request:
Using plane.nav_pitch_cd resulted in motors tilting up when flying level in Q modes.
It might also be a good idea to disable this when in flight to avoid the possibility of inducing control instability during attitude upsets. But is_flying() doesn't seem to be the right test...

IamPete1 commented:
LGTM, i will see if i can work out whats going on with my TVBS bronco in realflight and test this tommorow.

kd0aij commented:
This is still WIP because the motors don't tilt up until I raise the throttle in SITL. This would likely result in prop strikes with real vehicles.

IamPete1 commented:
I have just checked this again in SITL, tilts seems to always point up for me as they should

tridge commented:
nav_pitch_cd should have been correct here, I'd like to know why it wasn't

kd0aij commented:
@tridge @IamPete1 I found the root cause of the nav_pitch_cd problem and fixed it with a minimal change.
While I was testing, I also fixed a logging issue with bodyframe roll modes and added a constraint to protect from bad parameter values.
The motors tilt up reliably now for me in SITL/RF8/QHOVER, but they flop around a lot during bootup. Shouldn't we be suppressing servo outputs until init is complete?

kd0aij commented:
@tridge I just noticed that ACRO mode has the same issue with not setting nav_pitch/roll.
Of course,  if you are flying in pure rate mode, there is no attitude target. but it is easy to define a "virtual" attitude target calculated from the current attitude and the desired rates. This is what the target attitude quaternion represents in the multicopter controller.
